pyproject.toml

```diff
@@ -1,15 +1,82 @@
 [build-system]
-build-backend = "setuptools.build_meta"
+build-backend = "mesonpy"
 requires = [
-    "setuptools>=42",
-    "setuptools_scm[toml]>=7.0.0",
+    "meson-python",
     "pybind11>=2.8.0",
     "wheel",
+    "versioningit",
 ]
 
-[tool.setuptools_scm]
-version_scheme = "no-guess-dev"
-version_file = "pyamg/version.py"
+[project]
+name = "pyamg"
+dynamic = ["version"]
+authors = [
+    {name = "Nathan Bell"},
+    {name = "Luke Olson", email = "luke.olson@gmail.com"},
+    {name = "Jacob Schroder"},
+    {name = "Ben Southworth"},
+]
+maintainers = [
+    {name = "Luke Olson", email = "luke.olson@gmail.com"},
+    {name = "Jacob Schroder"},
+    {name = "Ben Southworth"},
+]
+description = "PyAMG: Algebraic Multigrid Solvers in Python"
+readme = "README.md"
+license = {text = "MIT License"}
+keywords = ["algebraic multigrid", "AMG", "sparse matrix", "preconditioning"]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Education",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Natural Language :: English",
+    "Operating System :: OS Independent",
+    "Programming Language :: C++",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: Education",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Mathematics",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+dependencies = [
+    "numpy",
+    "scipy>=1.11.0"
+]
```

Comment:
Looks good. Matches what we discussed in issue: https://github.com/pyamg/pyamg/issues/416